devlib/target.py

```diff
@@ -684,23 +687,42 @@ def sleep(self, duration):
         timeout = duration + 10
         self.execute('sleep {}'.format(duration), timeout=timeout)
 
-    def read_tree_values_flat(self, path, depth=1, check_exit_code=True):
-        command = 'read_tree_values {} {}'.format(quote(path), depth)
+    def read_tree_values_flat(self, path, depth=1, check_exit_code=True, decode_unicode=True):
+        command = 'read_tree_tgz_b64 {} {} {}'.format(quote(path), depth, self.working_directory)
         output = self._execute_util(command, as_root=self.is_rooted,
                                     check_exit_code=check_exit_code)
 
         accumulator = defaultdict(list)
-        for entry in output.strip().split('\n'):
-            if ':' not in entry:
-                continue
-            path, value = entry.strip().split(':', 1)
-            accumulator[path].append(value)
+
+        # Unpack the archive in memory
+        tar_gz = base64.b64decode(output)
+        tar_gz_bytes = io.BytesIO(tar_gz)
+        tar_buf = gzip.GzipFile(fileobj=tar_gz_bytes).read()
+        tar_bytes = io.BytesIO(tar_buf)
+        with tarfile.open(fileobj=tar_bytes) as tar:
+            for member in tar.getmembers():
+                try:
+                    content_f = tar.extractfile(member)
+                # ignore exotic members like sockets
+                except Exception:
+                    continue
+                # if it is a file and not a folder
+                if content_f:
+                    content = content_f.read()
+                    if decode_unicode:
+                        try:
+                            content = str(content.decode('utf-8').strip())
+                        except:
```

Comment:
except must always catch a specific exception, or at least a `Exception` class. Here I assume we would need to catch `UnicodeError`.

With a plain "except:", KeyboardInterrupt will be caught and hitting "ctrl-c" when that code is running will make it return an empty string, instead of terminating the process for example [1].

[1] https://www.python.org/dev/peps/pep-0008/#programming-recommendations